### PR TITLE
GEN-1277 - feat: added AverageRatingBlock

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

Screenshot 2023-10-09 at 10 03 52

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3350** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3350" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3350?utm_source=stack-comment).

File: .prettierignore [hunk 1]
```diff
@@ -4,4 +4,6 @@ coverage
 dist
 storybook-static
 
-generated.ts
\ No newline at end of file
+generated.ts
+
+mockServiceWorker.js
```
Comment (guilhermespopolin) [inline .prettierignore:9]: Added this file here since we probably don't want to format it.

File: apps/store/src/blocks/AverageRatingBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+type Product = ReturnType<typeof useProductPageContext>['productData']
+type AverageRating = NonNullable<ReturnType<typeof useProductPageContext>['averageRating']>
+
+export const AverageRatingBlock = () => {
+  const { productData, averageRating } = useProductPageContext()
+
+  if (!averageRating) {
+    console.warn(
+      `AverageRatingBlock | could not found average rating data for product ${productData.name}.`,
```
Comment (robinandeer) [inline apps/store/src/blocks/AverageRatingBlock.tsx:21]: nit: "AverageRatingBlock | Average rating data for product ${productData.name} not found."

File: apps/store/src/blocks/AverageRatingBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    )
+
+    return null
+  }
+
+  return (
+    <>
+      {isBrowser() && (
```
Comment (robinandeer) [inline apps/store/src/blocks/AverageRatingBlock.tsx:29]: why do we need this?

File: apps/store/src/blocks/AverageRatingBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                <TooltipPrimitive.Root delayDuration={0}>
+                  <TooltipPrimitive.Trigger asChild={true}>
+                    <button style={{ marginTop: 2 }}>
+                      <InfoIcon color={theme.colors.textSecondary} />
+                    </button>
+                  </TooltipPrimitive.Trigger>
+                  <TooltipPrimitive.Portal>
+                    <Content sideOffset={5} onClick={(event) => event.stopPropagation()}>
+                      <Text size="xs" color="textNegative" align="center">
+                        These ratings are obtained from users that have made a claim
```
Comment (robinandeer) [inline apps/store/src/blocks/AverageRatingBlock.tsx:72]: lokalise

File: apps/store/src/services/averageRating/averageRating.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { kv } from '@vercel/kv'
```
Comment (robinandeer) [inline apps/store/src/services/averageRating/averageRating.ts:1]: I wonder if this is it's own service or if it belongs to Trustpilot or Vercel?

**Review by robinandeer — APPROVED** [review]
Nice work :)))

Added via Giphy

File: apps/store/src/blocks/AverageRatingBlock.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+              </TooltipPrimitive.Provider>
+            </SpaceFlex>
+          </Wrapper>
+        </GridLayout.Content>
+      </GridLayout.Root>
+    </>
+  )
+}
+
+AverageRatingBlock.blockName = 'averageRating'
```
Comment (gustaveen) [inline apps/store/src/blocks/AverageRatingBlock.tsx:87]: nit: should we make the name more tied to the product, like ProductRatingBlock? :)

File: apps/store/src/services/averageRating/averageRating.ts (hunk 5 above)
Comment (gustaveen) [reply apps/store/src/services/averageRating/averageRating.ts:1]: Cool that you used KV, would love to learn more about how you implemented this @guilhermespopolin :)

**Review by gustaveen — APPROVED** [review]
Great job! 🥳

File: apps/store/src/services/averageRating/averageRating.ts (hunk 5 above)
Comment (guilhermespopolin) [reply apps/store/src/services/averageRating/averageRating.ts:1]: It "belongs" to Vercel. I mean, it's a [partnership](https://vercel.com/docs/storage/vercel-kv#upstash-partnership) between Vercel and Upstash. Through them, Vercel offers redis DB as storage option. We don't need to have a upstash account and the DB management is done with vercel dashboard.

File: apps/store/src/blocks/AverageRatingBlock.tsx (hunk 6 above)
Comment (guilhermespopolin) [reply apps/store/src/blocks/AverageRatingBlock.tsx:87]: Makse sense.

**Pull request merged**